Simplenote/Classes/SPNoteEditorViewController+Extensions.swift

```diff
@@ -616,6 +616,67 @@ extension SPNoteEditorViewController {
     }
 }
 
+// MARK: - Search Map
+//
+extension SPNoteEditorViewController {
+
+    /// Show search map keyword ranges
+    ///
+    @objc
+    func showSearchMap(with searchRangeValues: [NSValue]) {
+        createSearchMapViewIfNeeded()
+        // trigger layout to get correct height
+        searchMapView.superview?.layoutSubviews()
+
+        let editorRect = noteEditorTextView.layoutManager.usedRect(for: noteEditorTextView.textContainer)
+        let editorHeight = max(editorRect.size.height,
+                               searchMapView?.frame.size.height ?? 0)
+
+        guard editorHeight > CGFloat.leastNormalMagnitude else {
+            searchMapView?.update(with: [])
+            return
+        }
+
+        // Value from 0 to 1
+        let searchBarPositions: [CGFloat] = searchRangeValues.map {
+            let boundingRect = noteEditorTextView.boundingRectForCharacterRange($0.rangeValue)
+            return max(boundingRect.midY / editorHeight, CGFloat.leastNormalMagnitude)
+        }
+
+        searchMapView?.update(with: searchBarPositions)
```

Comment:
Idea: How about splitting this API into smaller methods? 

1. Editor Height calculation (potentially a UITextView extension). Maybe [this could help?](https://github.com/Automattic/simplenote-ios/blob/develop/Simplenote/Classes/UITextView+Simplenote.swift#L160)

2. Range + Height > Position conversion. Potentially another UITextView extension?

This approach would allow you to return `[]` whenever `editorHeight > CGFloat.leastNormalMagnitude`, and you could kill the guard from `L635`.

Reply:
1. Var name was confusing (renamed it to `textContainerHeight`), it's actually rect for a text container, because I'm interested in the total height of the content of the text container. It is quite specific to the search map, so I wouldn't put in a UITextView extension.
2. I don't think it's good to put it in an extension as it is very specific to the search map. 

I split it in a several methods, but kept them in the `SPNoteEditorViewController`. Thanks for the suggestion 👍